File: problems/283. Move Zeroes/solution.py
class Solution:
    
    def moveZeroes(self, nums):
        slot = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[slot] = nums[i]
                slot += 1
        for j in range(slot, len(nums)):
            nums[j] = 0

File: problems/283. Move Zeroes/solution.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int slot = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i])
            {
                swap(slot, i, nums);
                slot++;
            }
        }
    }

    void swap(int i, int j, vector<int> &array)
    {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
};